cpp/libcudf_kafka/CMakeLists.txt

```diff
@@ -35,11 +35,13 @@ endif()
 ###################################################################################################
 # - Build options
 
+option(CUDF_USE_ARROW_STATIC "Build static arrow library" OFF)
```

Comment:
Do we need to define this option for `libcudf_kafka` or can someone still just pass `-DCUDF_USE_ARROW_STATIC` to pass into the libcudf build regardless?

Reply:
I wasn't sure what CPM's behavior here was, so for safety I define it here and pass it down in the `CPMFindPackage` options. We can remove it if it's automatically propagated to the libcudf build.

Reply:
No, this is no longer needed as you stated. I will remove it

Reply:
Ok, apparently this is still needed as the build fails when I remove it so I'm adding it back.

Reply:
This sets a bad precedent of all build options needing to be completely transitive. There has to be a cleaner way of handling this, no?

Reply:
I am unsure. @trxcllnt do you know?

Reply:
I'm not sure why it fails without this, but I've seen the failure as well. Seems like `CUDF_USE_ARROW_STATIC` isn't defined [here](https://github.com/rapidsai/cudf/blob/branch-0.19/cpp/cmake/thirdparty/CUDF_GetArrow.cmake#L125) unless we add it to our `CMakeLists.txt` and pass it down here @robertmaynard any ideas?

Reply:
You most likely need a clean build directory.

If you are running `cmake -DCUDF_USE_ARROW_STATIC=ON -B <build_dir>` the `CUDF_USE_ARROW_STATIC` variables will be placed into the `CMakeCache` before parsing/execution of the project CMake code. After that it should be visible through the entire execution of CMake.

Reply:
So seems like we have a clean build dir looking at https://github.com/rapidsai/cudf/blob/branch-0.19/build.sh#L194 does the `CUDF_USE_ARROW_STATIC` variable simply need to be added here for this to work then?

Reply:
It does @jdye64 when arrow is static.

Reply:
ok, I have added that to `build.sh` then and removed the option from the `libcudf_kafka` CMakeLists.txt file.